Source/Core/InputCommon/ControllerInterface/Quartz/QuartzKeyboardAndMouse.h

```diff
@@ -59,13 +59,16 @@ class KeyboardAndMouse : public Core::Device
   void UpdateInput() override;
 
   explicit KeyboardAndMouse(void* view);
+  ~KeyboardAndMouse() override;
 
   std::string GetName() const override;
   std::string GetSource() const override;
 
 private:
+  void MainThreadInitialization(void* view);
+
   Common::Vec2 m_cursor;
 
-  uint32_t m_windowid;
+  void* m_window_pos_observer;
```

Comment:
If I'm not mistaken, defining this as `void*` requires us to do the odd bridge stuff in the `mm` to make sure ARC counts references properly. Would it be better to just leave it as an Objective-C type so we don't need to do that? (Might need to add `typedef void DolWindowPositionObserver` in an `#ifndef __OBJC__` to make C++ only source files happy, though.)

Reply:
[Done](https://github.com/dolphin-emu/dolphin/compare/55828fa4c462ebdb10cf4121d58c48e988aaa0ee..588c4bd6355f20dd74008054b1bb876810f4ba13)